<?xml version="1.0"?>
<robot name="sobit">

<material name="black">
        <color rgba="0.3 0.3 0.3 0.8"/>
      </material>

<material name="black2">
        <color rgba="0.1 0.1 0.1 1.0"/>
      </material>

  <!--link ####################################################################-->
  <!--base_frame link-->
  <link name="base_frame">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_MainFrame v1.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.06"/>
      <material name="black2">
      </material>
    </visual>
  </link>

  <!--left_wheel link-->
  <link name="left_wheel">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/wheel.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--right_wheel link-->
  <link name="right_wheel">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/wheel.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--right_shoulder link-->
  <link name="right_shoulder">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_shoulder v1.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--left_shoulder link-->
  <link name="left_shoulder">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_shoulder v1.stl"/>
      </geometry>
      <origin rpy="0 0 3.14" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--right_First_arm link-->
  <link name="right_first_arm">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/first_arm.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.17"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--left_First_arm link-->
  <link name="left_first_arm">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/first_arm.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.17"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--right_elbow link-->
  <link name="right_elbow">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_elbow v2.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--left_elbow link-->
  <link name="left_elbow">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_elbow v2.stl"/>
      </geometry>
      <origin rpy="0 0 3.14" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--right_second_arm link-->
  <link name="right_second_arm">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_second_arm v2.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--left_second_arm link-->
  <link name="left_second_arm">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_second_arm v2.stl"/>
      </geometry>
      <origin rpy="0 0 3.14" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>
   
  <!-- head -->
  <!--neck_link-->
  <link name="neck">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_neck.stl"/>
      </geometry>
      <origin rpy="0 0 1.517" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--chin_link-->
  <link name="chin">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_chin.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--head_link-->
  <link name="head">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_head.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!-- left_hand -->
  <!--left_hand_link-->
  <link name="left_hand">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_hand.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--left_thumb_link-->
  <link name="left_hand_thumb">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_thumb.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--left_index_link-->
  <link name="left_hand_index">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_index.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--left_middle_link-->
  <link name="left_hand_middle">
    <visual>
      <geometry>
         <mesh filename="package://sobit_description/mesh/SOBIT_index.stl" scale="1 1 1.0588"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--left_ring_link-->
  <link name="left_hand_ring">
    <visual>
      <geometry>
         <mesh filename="package://sobit_description/mesh/SOBIT_index.stl" scale="1 1 1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--left_pinky_link-->
  <link name="left_hand_pinky">
    <visual>
      <geometry>
         <mesh filename="package://sobit_description/mesh/SOBIT_index.stl" scale="1 1 0.88"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!-- right_hand -->
  <!--right_hand_link-->
  <link name="right_hand">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_hand_right.stl"/>
      </geometry>
      <origin rpy="0 0 3.14" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

   <!--right_thumb_link-->
  <link name="right_hand_thumb">
    <visual>
      <geometry>
        <mesh filename="package://sobit_description/mesh/SOBIT_thumb_right.stl"/>
      </geometry>
      <origin rpy="0 3.14 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>

  <!--right_hand_twist link-->
  <link name="right_hand_twist">
    <visual>
      <geometry>
        <box size="0.06 0.01 0.07"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.03 0.01 -0.035"/>
      <material name="black">
      </material>
    </visual>
  </link>


  <!--lidar link-->
  <link name="lidar">
    <visual>
      <geometry>
        <cylinder length="0.05" radius="0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black">
      </material>
    </visual>
  </link>










  <!-- joint  ################################################################## -->
  <!-- wheel -->
  <!--left_wheel joint-->
  <joint name="L_wheel" type="continuous">
    <parent link="base_frame"/>
    <child link="left_wheel"/>
    <axis xyz="0 1 0"/>
    <origin xyz="0.08 0.16 0.05"/>
  </joint>

  <!--right_wheel joint-->
  <joint name="R_wheel" type="continuous">
    <parent link="base_frame"/>
    <child link="right_wheel"/>
    <axis xyz="0 1 0"/>
    <origin xyz="0.08 -0.16 0.05"/>
  </joint>

  <!-- left_arm -->
  <!--left_shoulder to left_first_arm joint-->
  <joint name="L_shoulder_roll" type="revolute">
    <parent link="left_shoulder"/>
    <child link="left_first_arm"/>
    <axis xyz="1 0 0"/>
    <origin rpy="0.0 0 0" xyz="0.00 0.075 0"/>
     <limit effort="1000" lower="0" upper="3.14" velocity="0.5"/>
  </joint>

  <!--left_shoulder_pitch joint-->
  <joint name="L_shoulder_pitch" type="revolute">
    <parent link="base_frame"/>
    <child link="left_shoulder"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 6.0" xyz="0.05 0.15 0.79"/>
     <limit effort="1000" lower="-3.14" upper="1.571" velocity="0.5"/>
  </joint>

  <!--left_elbow_yaw joint-->
  <joint name="L_elbow_yaw" type="revolute">
    <parent link="left_first_arm"/>
    <child link="left_elbow"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 -1.571" xyz="0.00 0 -0.175"/>
     <limit effort="1000" lower="-1.571" upper="3.14" velocity="0.5"/>
  </joint>

  <!--left_elbow to left_second_arm joint-->
  <joint name="L_elbow_pitch" type="revolute">
    <parent link="left_elbow"/>
    <child link="left_second_arm"/>
    <axis xyz="1 0 0"/>
    <origin rpy="0 0 0" xyz="0.00 -0.055 -0.045"/>
     <limit effort="1000" lower="-3.14" upper="0.3" velocity="0.5"/>
  </joint>
 
  <!-- right_arm -->
  <!--right_shoulder to right_first_arm joint-->
  <joint name="R_shoulder_roll" type="revolute">
    <parent link="right_shoulder"/>
    <child link="right_first_arm"/>
    <axis xyz="1 0 0"/>
    <origin rpy="0.0 0 0" xyz="0.00 -0.075 0"/>
     <limit effort="1000" lower="-3.14" upper="0" velocity="0.5"/>
  </joint>

  <!--shoulder_pitch-->
  <joint name="R_shoulder_pitch" type="revolute">
    <parent link="base_frame"/>
    <child link="right_shoulder"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0.27" xyz="0.05 -0.15 0.79"/>
     <limit effort="1000" lower="-3.14" upper="1.571" velocity="0.5"/>
  </joint>

  <!--elbow_yaw joint-->
  <joint name="R_elbow_yaw" type="revolute">
    <parent link="right_first_arm"/>
    <child link="right_elbow"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 1.571" xyz="0.00 0 -0.175"/>
     <limit effort="1000" lower="-3.14" upper="1.571" velocity="0.5"/>
  </joint>

  <!--elbow_pitch joint-->
  <joint name="R_elbow_pitch" type="revolute">
    <parent link="right_elbow"/>
    <child link="right_second_arm"/>
    <axis xyz="1 0 0"/>
    <origin rpy="0 0 0" xyz="0.00 0.055 -0.045"/>
     <limit effort="1000" lower="-0.3" upper="3.14" velocity="0.5"/>
  </joint>

  <!-- head -->
  <!--neck pitch joint-->
  <joint name="neck_pitch" type="revolute">
    <parent link="neck"/>
    <child link="chin"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0 0.11"/>
     <limit effort="1000" lower="-0.505" upper="0.505" velocity="0.5"/>
  </joint>

  <!--neck yaw joint-->
  <joint name="neck_yaw" type="revolute">
    <parent link="chin"/>
    <child link="head"/>
    <axis xyz="1 0 0"/>
    <origin rpy="0 0 0" xyz="-0.016 0 0"/>
     <limit effort="1000" lower="-0.505" upper="0.505" velocity="0.5"/>
  </joint>

  <!--neck roll-->
  <joint name="neck_roll" type="revolute">
    <parent link="base_frame"/>
    <child link="neck"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0 0 0.86"/>
     <limit effort="1000" lower="-1.571" upper="1.571" velocity="0.5"/>
  </joint>


  <!-- left_hand -->
  <!--left_hand_wrist joint-->
  <joint name="L_hand_wrist" type="revolute">
    <parent link="left_second_arm"/>
    <child link="left_hand"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 1.571" xyz="0 0.04 -0.17"/>
     <limit effort="1000" lower="-1.571" upper="1.571" velocity="0.5"/>
  </joint>

  <!--left_hand_thumb joint-->
  <joint name="L_hand_thumb" type="revolute">
    <parent link="left_hand"/>
    <child link="left_hand_thumb"/>
    <axis xyz="0 -1 0"/>
    <origin rpy="0 0 0" xyz="0.04 -0.035 -0.055"/>
     <limit effort="1000" lower="-1.571" upper="1.571" velocity="0.5"/>
  </joint>

  <!--left_hand_index joint-->
  <joint name="left_hand_index" type="revolute">
    <parent link="left_hand"/>
    <child link="left_hand_index"/>
    <axis xyz="1 0 0"/>
    <origin rpy="0 0 0" xyz="0.04 -0.01 -0.09"/>
     <limit effort="1000" lower="-1.571" upper="0" velocity="0.5"/>
  </joint> 

  <!--left_hand_middle joint-->
  <joint name="L_hand_middle" type="revolute">
    <parent link="left_hand"/>
    <child link="left_hand_middle"/>
    <axis xyz="1 0 0"/>
    <origin rpy="0 0 0" xyz="0.02 -0.01 -0.09"/>
     <limit effort="1000" lower="-1.571" upper="0" velocity="0.5"/>
  </joint> 

  <!--left_hand_ring joint-->
  <joint name="L_hand_ring" type="revolute">
    <parent link="left_hand"/>
    <child link="left_hand_ring"/>
    <axis xyz="1 0 0"/>
    <origin rpy="0 0 0" xyz="0 -0.01 -0.09"/>
     <limit effort="1000" lower="-1.571" upper="0" velocity="0.5"/>
  </joint> 

  <!--left_hand_pinky joint-->
  <joint name="L_hand_pinky" type="revolute">
    <parent link="left_hand"/>
    <child link="left_hand_pinky"/>
    <axis xyz="1 0 0"/>
    <origin rpy="0 0 0" xyz="-0.02 -0.01 -0.09"/>
     <limit effort="1000" lower="-1.571" upper="0" velocity="0.5"/>
  </joint> 

  <!-- right_hand -->
  <!--right_hand_wrist joint-->
  <joint name="right_arm_to_right_hand" type="fixed">
    <parent link="right_second_arm"/>
    <child link="right_hand"/>
    <origin rpy="0 0 1.571" xyz="0 -0.04 -0.17"/>  
  </joint>

  <!--right_hand_wrist joint-->
  <joint name="right_hand_to_right_thumb" type="fixed">
    <parent link="right_hand"/>
    <child link="right_hand_thumb"/>
    <origin rpy="0 0 0" xyz="-0.04 -0.035 -0.055"/>
  </joint> 

  <!--right_hand_twist joint-->
  <joint name="R_hand_twist" type="revolute">
    <parent link="right_hand"/>
    <child link="right_hand_twist"/>
    <axis xyz="1 0 0"/>
    <origin rpy="0 0 0" xyz="-0.02 -0.01 -0.09"/>
     <limit effort="1000" lower="-1.571" upper="0" velocity="0.5"/>
  </joint>


  <!-- lidar -->
  <!--lidar joint-->
  <joint name="lidar" type="fixed">
    <parent link="base_frame"/>
    <child link="lidar"/>
    <origin rpy="0 0 3.14" xyz="0.2 0 0.05"/>
  </joint> 

</robot>

